src/com/shashi/service/impl/UserServiceImpl.java

```diff
@@ -239,4 +242,121 @@ public String getUserAddr(String userId) {
 		return userAddr;
 	}
 
+
+
+	@Override
+	public String registerSellerUser(SellerBean user) {
+		String status = "Company Registration Failed!";
+
+		boolean isRegtd = isRegisteredSeller(user.getEmail(), user.getCompanyName());
+
+		if (isRegtd) {
+			status = "Email Id or Company Already Registered!";
+			return status;
+		}
+		Connection conn = DBUtil.provideConnection();
+		PreparedStatement ps = null;
+		if (conn != null) {
+			System.out.println("Connected Successfully!");
+		}
+
+		try {
+
+			ps = conn.prepareStatement("insert into " + "seller" + " values(?,?,?,?,?,?,?)");
+
+			ps.setString(1, user.getEmail());
+			ps.setString(2, user.getName());
+			ps.setLong(3, user.getMobile());
+			ps.setString(4, user.getAddress());
+			ps.setInt(5, user.getPinCode());
+			ps.setString(6, user.getPassword());
+			ps.setString(7, user.getCompanyName());
+
+			int k = ps.executeUpdate();
+
+			if (k > 0) {
+				status = "Company Registered Successfully!";
+				MailMessage.registrationSuccess(user.getEmail(), user.getName().split(" ")[0]);
+			}
+
+		} catch (SQLException e) {
+			status = "Error: " + e.getMessage();
+			e.printStackTrace();
+		}
+
+		DBUtil.closeConnection(ps);
+		DBUtil.closeConnection(ps);
+
+		return status;
+	}
+
+
+
+	@Override
+	public boolean isRegisteredSeller(String emailId, String companyName) {
+		boolean flag = false;
+
+		Connection con = DBUtil.provideConnection();
+
+		PreparedStatement ps = null;
+		ResultSet rs = null;
+
+		try {
+			ps = con.prepareStatement("select * from seller where email=? or companyName=?");
+
+			ps.setString(1, emailId);
+			ps.setString(2, companyName);
+
+			rs = ps.executeQuery();
+
+			if (rs.next())
+				flag = true;
+
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+
+		DBUtil.closeConnection(con);
+		DBUtil.closeConnection(ps);
+		DBUtil.closeConnection(rs);
+
+		return flag;
+	}
+
+
+
+	@Override
+	public String isValidCredentialSeller(String emailId, String password, String companyName) {
```

Comment:
this might be a bit difficult to do, but you are adding a lot of duplicated code here. Is there really no better way to achieve the same goal here?

Reply:
I am thinking of doing ifs inside the implementation of the original function depending on the userType (which we could pass as an additional parameter).

Reply:
sure cleaner code is always better